components/mediation/tasks/org.wso2.micro.integrator.ntask.core/src/main/java/org/wso2/micro/integrator/ntask/coordination/task/db/TaskQueryHelper.java

```diff
@@ -0,0 +1,76 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.micro.integrator.ntask.coordination.task.db;
+
+import org.wso2.micro.integrator.ntask.coordination.task.ConstantUtil;
+import org.wso2.micro.integrator.ntask.coordination.task.CoordinatedTask;
+
+/**
+ * The class which contains all the data base queries for the task database.
+ */
+public class TaskQueryHelper {
+
+    public static final String ADD_TASK =
+            "INSERT INTO " + ConstantUtil.TABLE_NAME + " ( " + ConstantUtil.TASK_NAME + ", "
+                    + ConstantUtil.DESTINED_NODE_ID + ", " + ConstantUtil.TASK_STATE + ") " + "VALUES (?,NULL,?)";
+
+    public static final String UPDATE_TASK =
+            "UPDATE  " + ConstantUtil.TABLE_NAME + " SET  " + ConstantUtil.DESTINED_NODE_ID + " = ? , "
+                    + ConstantUtil.TASK_STATE + " = ?  WHERE " + ConstantUtil.TASK_NAME + " = ?";
+
+    public static final String UPDATE_TASK_STATE =
+            "UPDATE  " + ConstantUtil.TABLE_NAME + "  SET " + ConstantUtil.TASK_STATE + " = ? WHERE "
+                    + ConstantUtil.TASK_NAME + " =? ";
+
+    public static final String RETRIEVE_ALL_TASKS =
+            "SELECT  " + ConstantUtil.TASK_NAME + " FROM " + ConstantUtil.TABLE_NAME;
+
+    public static final String RETRIEVE_UNASSIGNED_NOT_COMPLETED_TASKS =
+            "SELECT * FROM " + ConstantUtil.TABLE_NAME + " WHERE  " + ConstantUtil.DESTINED_NODE_ID + " IS NULL AND "
+                    + ConstantUtil.TASK_STATE + " !=\"" + CoordinatedTask.States.COMPLETED + "\"";
+
+    public static final String RETRIEVE_TASKS_OF_PARTICULAR_NODE =
+            "SELECT " + ConstantUtil.TASK_NAME + " FROM " + ConstantUtil.TABLE_NAME + "  WHERE "
+                    + ConstantUtil.DESTINED_NODE_ID + " =? AND " + ConstantUtil.TASK_STATE + " =?";
+
+    public static final String CLEAN_INVALID_NODES =
+            "UPDATE " + ConstantUtil.TABLE_NAME + " SET " + ConstantUtil.DESTINED_NODE_ID + " = NULL , "
+                    + ConstantUtil.TASK_STATE + " = ? WHERE " + ConstantUtil.TASK_NAME + " =?";
+
+    public static final String REMOVE_TASKS_OF_NODE =
+            "DELETE FROM " + ConstantUtil.TABLE_NAME + "  WHERE " + ConstantUtil.DESTINED_NODE_ID + " =?";
+
+    public static final String DELETE_TASK =
+            "DELETE FROM " + ConstantUtil.TABLE_NAME + " WHERE " + ConstantUtil.TASK_NAME + " =?";
+
+    public static final String CLEAN_TASKS_OF_NODE =
+            "UPDATE " + ConstantUtil.TABLE_NAME + " SET " + ConstantUtil.DESTINED_NODE_ID + " = NULL , "
```

Comment:
Should we set the DESTINED_NODE_ID to null even when the task are not in a running state. e.g., if the task was completed, we might loose information as to who completed it, whether there was a bug, etc.

Reply:
This query will set the state to none only for running tasks not others.